[mqtt] Homie doesn't accept properties with no $name attribute

Hi @davidgraeff!

I have seen the following error in the openhab.log file when adding a new Homie device to my openHAB installation:

```
2019-03-21 17:04:57.578 [INFO ] [g.discovery.internal.PersistentInbox] - Added new thing 'mqtt:homie300:embedded-mqtt-broker:aiturrioz-relay' to inbox.
2019-03-21 17:05:04.201 [WARN ] [ic.internal.convention.homie300.Node] - Could not subscribe
java.util.concurrent.CompletionException: java.lang.Exception: Did not receive mandatory topic value: homie/aiturrioz-relay/switch/on/$name
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308) ~[?:?]
	at java.util.concurrent.CompletableFuture.biRelay(CompletableFuture.java:1284) ~[?:?]
	at java.util.concurrent.CompletableFuture$BiRelay.tryFire(CompletableFuture.java:1270) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977) ~[?:?]
	at org.openhab.binding.mqtt.generic.internal.mapping.SubscribeFieldToMQTTtopic.timeoutReached(SubscribeFieldToMQTTtopic.java:161) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
Caused by: java.lang.Exception: Did not receive mandatory topic value: homie/aiturrioz-relay/switch/on/$name
	... 8 more
```
Here you can see the list of MQTT messages sent by the device:

```
homie/aiturrioz-relay/$stats/signal 44
homie/aiturrioz-relay/switch/$type switch
homie/aiturrioz-relay/switch/on/$settable true
homie/aiturrioz-relay/$fw/version 1.0.0
homie/aiturrioz-relay/$state ready
homie/aiturrioz-relay/$stats/uptime 0
homie/aiturrioz-relay/$implementation homie-python
homie/aiturrioz-relay/switch/$name switch
homie/aiturrioz-relay/$homie 3.0.1
homie/aiturrioz-relay/$nodes switch
homie/aiturrioz-relay/switch/$properties on
homie/aiturrioz-relay/$name Relay AIturrioz
homie/aiturrioz-relay/$stats/interval 60
homie/aiturrioz-relay/$mac 00:00:00:00:00:00
homie/aiturrioz-relay/$fw/name relay-switch
homie/aiturrioz-relay/$localip 127.0.0.1
```
As far as I know ([Homie Convention](https://homieiot.github.io/specification/#property-attributes)), the $name attribute is not a required one for the properties, so there is a bug in the implementation.

Many thanks for your great support and amazing work!

Aitor

--- a/addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homie300/Property.java
+++ b/addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homie300/Property.java
@@ -160,6 +160,11 @@ public class Property implements AttributeChanged {
 
         Value value;
         Boolean isDecimal = null;
+
+        if (attributes.name == "") {
+            attributes.name = propertyID;
+        }
+
         switch (attributes.datatype) {
             case boolean_:
                 value = new OnOffValue("true", "false");

--- a/addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homie300/PropertyAttributes.java
+++ b/addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homie300/PropertyAttributes.java
@@ -18,7 +18,6 @@ import java.util.TreeMap;
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.openhab.binding.mqtt.generic.internal.mapping.AbstractMqttAttributeClass;
 import org.openhab.binding.mqtt.generic.internal.mapping.MQTTvalueTransform;
-import org.openhab.binding.mqtt.generic.internal.mapping.MandatoryField;
 import org.openhab.binding.mqtt.generic.internal.mapping.TopicPrefix;
 
 /**
@@ -40,7 +39,7 @@ public class PropertyAttributes extends AbstractMqttAttributeClass {
         color_
     }
 
-    public @MandatoryField String name = "";
+    public String name = "";
 
     /**
      * stateful + non-settable: The node publishes a property state (temperature sensor)
@@ -53,7 +52,7 @@ public class PropertyAttributes extends AbstractMqttAttributeClass {
     public boolean settable = false;
     public boolean retained = true;
     public String unit = "";
-    public @MandatoryField @MQTTvalueTransform(suffix = "_") DataTypeEnum datatype = DataTypeEnum.unknown;
+    public @MQTTvalueTransform(suffix = "_") DataTypeEnum datatype = DataTypeEnum.unknown;
     public String format = "";
 
     @Override

